### PR TITLE
HPCC-14160 Cache package dependencies

### PR DESCRIPTION
**Pull request by Michael-Gardner**
This change allows for us to create package dependencies that can be passed between projects.  

Signed-off-by: Michael Gardner Michael.Gardner@lexisnexis.com

### CODE REVIEW

**Comment by Michael-Gardner** [comment]
@xwang2713 Please review

**Comment by mckellyln** [comment]
Is there perhaps a way to not have to list the dependencies twice in every file ?

**Comment by Michael-Gardner** [comment]
yes we could do something like:
set( deplist "list, of, dependencies")
if ( CPACK_DEBIAN_PACKAGE_DEPENDS )
    set ( CPACK_DEBIAN_PACKAGE_DEPENDS "${CPACK_DEBIAN_PACKAGE_DEPENDS}, ${deplist}" CACHE STRING "" FORCE)
else()
    set (CPACK_DEBIAN_PACKAGE_DEPENDS "${deplist}" CACHE STRING "")
endif()

**Comment by Michael-Gardner** [comment]
@mckellyln @xwang2713 I added a macro to take care of appending to the cache variable.  To add a dependency now just do SET_DEPENDENCIES ( "comma, separated, dependencies" VAR_TO_SET )

File: cmake_modules/commonSetup.cmake [hunk 1]
```diff
@@ -739,4 +739,16 @@ IF ("${COMMONSETUP_DONE}" STREQUAL "")
         endif()
   ENDMACRO()
 
+  ###########################################################################
+  ###
+  ## The following sets the dependency list for a package
+  ###
+  MACRO(SET_DEPENDENCIES dependencies cpackvar)
```
Comment (mckellyln) [inline cmake_modules/commonSetup.cmake]: great job - minor question: can we reorder so cpackvar is first, to be more like it was ?

Comment (Michael-Gardner) [inline cmake_modules/commonSetup.cmake]: certainly.  That's a good idea too, that way it's more consistent with the way the built in set function works.

**Comment by Michael-Gardner** [comment]
I think I might have found a potential downside to this approach.  

if you run cmake initially, then make changes to the source, and run cmake again, the cache variable is still populated and will get dependencies added twice.  I need to either add a flag that persists between build attempts, or sanitize the list for duplicates.

**Comment by Michael-Gardner** [comment]
@mckellyln So I fixed the problem I brought up earlier today.  The easiest way to do it was to convert any value found in the CPACK dependency variable over to a list, then append any other dependencies onto it, and do a list (REMOVE_DUPLICATES), then convert it back to a string and update the cache variable.  Now if you configure multiple times you're still safe.  I'll get the updated commit submitted shortly.

File: cmake_modules/dependencies/utopic.cmake [hunk 2]
```diff
@@ -1 +1 @@
-set ( CPACK_DEBIAN_PACKAGE_DEPENDS "libboost-regex1.55.0, libicu52, libxslt1.1, libxml2, binutils, libldap-2.4-2, openssl, zlib1g, g++, openssh-client, openssh-server, expect, libarchive13, rsync, libapr1, libaprutil1, python, libtbb2")
+SET_DEPENDENCIES ( CPACK_DEBIAN_PAcKAGE_DEPENDS libboost-regex1.55.0 libicu52 libxslt1.1 libxml2 binutils libldap-2.4-2 openssl zlib1g g++ openssh-client openssh-server expect libarchive13 rsync libapr1 libaprutil1 python libtbb2 )
```
Comment (richardkchapman) [inline cmake_modules/dependencies/utopic.cmake]: Why the lower case c ?

Comment (Michael-Gardner) [inline cmake_modules/dependencies/utopic.cmake]: Fixing it now.  Finger missed the shift key on that one.

**Comment by mckellyln** [comment]
review looks ok to merge.

**Pull request merged**